Framework/Muon/src/MuonAlgorithmHelper.cpp

```diff
@@ -516,5 +516,86 @@ bool checkValidGroupPairName(const std::string &name) {
   return true;
 }
 
+/**
+ * Sums the specified periods of the input workspace group
+ * @param periodsToSum :: [input] List of period indexes (1-based) to be summed
+ * @returns Workspace containing the sum
+ */
+MatrixWorkspace_sptr sumPeriods(const WorkspaceGroup_sptr &inputWS,
+                                const std::vector<int> &periodsToSum) {
+  MatrixWorkspace_sptr outWS;
+  if (!periodsToSum.empty()) {
+    auto LHSWorkspace = inputWS->getItem(periodsToSum[0] - 1);
+    outWS = boost::dynamic_pointer_cast<MatrixWorkspace>(LHSWorkspace);
+    if (outWS != nullptr && periodsToSum.size() > 1) {
+      int numPeriods = static_cast<int>(periodsToSum.size());
+      for (int i = 1; i < numPeriods; i++) {
+        auto RHSWorkspace = inputWS->getItem(periodsToSum[i] - 1);
+        IAlgorithm_sptr alg = AlgorithmManager::Instance().create("Plus");
+        alg->setChild(true);
+        alg->setProperty("LHSWorkspace", outWS);
+        alg->setProperty("RHSWorkspace", RHSWorkspace);
+        alg->setProperty("OutputWorkspace", "__NotUsed__");
+        alg->execute();
+        outWS = alg->getProperty("OutputWorkspace");
+      }
+    }
+  }
+  return outWS;
+}
+
+/**
+ * Subtracts one workspace from another: lhs - rhs.
+ * @param lhs :: [input] Workspace on LHS of subtraction
+ * @param rhs :: [input] Workspace on RHS of subtraction
+ * @returns Result of the subtraction
+ */
+MatrixWorkspace_sptr subtractWorkspaces(const MatrixWorkspace_sptr &lhs,
+                                        const MatrixWorkspace_sptr &rhs) {
+  MatrixWorkspace_sptr outWS;
```

Comment:
I think if it fails we should return the `LHSWorkspace` .i.e. subtract nothing.

Reply:
If we do this do we not risk hiding any error? Is there a case where we would want to carry on having failed to subtract rather than flagging up the failure?

Reply:
thats true. In which case we should say it failed. As far as I can tell it currently fails silently.

Reply:
I have added setRethrows(true) to the algorithms which should ensure that they propagate errors when something goes wrong. We will need to ensure that we catch these errors somewhere however.

Reply:
if we run this in a thread it will automatically catch it https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/thread_model.py and there is a PR for a more sophisticated version (but has bugs)